### PR TITLE
Feature/informal update models

### PR DESCRIPTION
**Pull request by Rup-Narayan-Rajbanshi**
## Changes

* Added new app informal_update
* Created related models
* Added fixtures for InformalActions

## Checklist 
Things that should succeed before merging.

- [ ] Updated/ran unit tests
- [ ] Updated CHANGELOG.md

## Release

If there is a version update, make sure to tag the repository with the latest version.

### CODE REVIEW

File: api/fixtures/Informal_actions.json [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    },
+    {
+      "model": "informal_update.InformalAction",
+      "pk": "41",
+      "fields": {
+        "name": "CVA/Basic needs"
+      }
+    }
+  ]
+
```
Comment (k9845) [inline api/fixtures/Informal_actions.json:108]: Lets keep this inside `informal_update`

File: api/models.py [hunk 2]
```diff
@@ -1377,11 +1376,13 @@ class ActionOrg:
     NATIONAL_SOCIETY = 'NTLS'
     FOREIGN_SOCIETY = 'PNS'
     FEDERATION = 'FDRN'
+    GOVERNMENT = 'GOV'
 
     CHOICES = (
         (NATIONAL_SOCIETY, _('National Society')),
         (FOREIGN_SOCIETY, _('Foreign Society')),
         (FEDERATION, _('Federation')),
+        (GOVERNMENT, _('Government')),
```
Comment (k9845) [inline api/models.py:1385]: Do we need this, for `informal_update`?

File: informal_update/models.py [hunk 3]
```diff
@@ -0,0 +1,151 @@
+from django.db import models
+from django.conf import settings
+from django.utils.translation import ugettext_lazy as _
+from django.contrib.postgres.fields import ArrayField
+
+from enumfields import IntEnum, EnumIntegerField
+from api.storage import get_storage
+
+
+# Create your models here.
```
Comment (k9845) [inline informal_update/models.py:10]:
```suggestion
```

File: informal_update/models.py [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+
+
+class InformalReferences(models.Model):
+    date = models.DateTimeField(verbose_name=_('date'), blank=True)
+    source_description = models.CharField(verbose_name=_('Name or Source Description'), max_length=225, blank=True)
+    url = models.ManyToManyField(ReferenceUrls, verbose_name=_('Add url'), blank=True)
+
+
+class InformalCountryDistrict(models.Model):
+    from api.models import Country, District
```
Comment (k9845) [inline informal_update/models.py:22]: Lets not import this here...only if we have circular import we import here

File: informal_update/models.py [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+
+    countries = models.ForeignKey(
+        Country, verbose_name=_('countries'), on_delete=models.CASCADE,
+        related_name='informal_country'
+    )
```
Comment (k9845) [inline informal_update/models.py:27]:
```suggestion
    country = models.ForeignKey(
        Country, verbose_name=_('country'), on_delete=models.CASCADE,
        related_name='informal_country'
    )
```

File: informal_update/models.py [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+    districts = models.ForeignKey(
+        District, verbose_name=_('districts'), on_delete=models.CASCADE,
+        related_name='informal_district'
+    )
```
Comment (k9845) [inline informal_update/models.py:31]:
```suggestion
    district = models.ForeignKey(
        District, verbose_name=_('district'), on_delete=models.CASCADE,
        related_name='informal_district'
    )
```

File: informal_update/models.py [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+    url = models.URLField()
+
+
+class InformalReferences(models.Model):
+    date = models.DateTimeField(verbose_name=_('date'), blank=True)
+    source_description = models.CharField(verbose_name=_('Name or Source Description'), max_length=225, blank=True)
+    url = models.ManyToManyField(ReferenceUrls, verbose_name=_('Add url'), blank=True)
+
+
+class InformalCountryDistrict(models.Model):
```
Comment (k9845) [inline informal_update/models.py:21]: may be add `unique_together=('country', 'district')`

File: informal_update/models.py [hunk 8]
```diff
@@ -0,0 +1,151 @@
...
+        Country, verbose_name=_('countries'), on_delete=models.CASCADE,
+        related_name='informal_country'
+    )
+    districts = models.ForeignKey(
+        District, verbose_name=_('districts'), on_delete=models.CASCADE,
+        related_name='informal_district'
+    )
+
+
+class ShareChoices(IntEnum):
```
Comment (k9845) [inline informal_update/models.py:34]:
```suggestion
class InformShareChoices(IntEnum):
```

Comment (k9845) [reply informal_update/models.py:34]: Also lets use django `models.TextChoices`

File: informal_update/models.py [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+    originator_email = models.CharField(verbose_name=_('email'), max_length=300)
+    originator_phone = models.CharField(verbose_name=_('phone'), max_length=50, blank=True)
+
+    ifrc_name = models.CharField(verbose_name=_('name'), max_length=100)
+    ifrc_title = models.CharField(verbose_name=_('title'), max_length=300)
+    ifrc_email = models.CharField(verbose_name=_('email'), max_length=300)
+    ifrc_phone = models.CharField(verbose_name=_('phone'), max_length=50, blank=True)
+
+    # Share with
+    visibility = EnumIntegerField(ShareChoices, verbose_name=_('share with'), default=1)
```
Comment (k9845) [inline informal_update/models.py:83]: lets not provide default here!!

File: informal_update/models.py [hunk 10]
```diff
@@ -0,0 +1,151 @@
...
+    # Focal Point
+    originator_name = models.CharField(verbose_name=_('name'), max_length=100)
+    originator_title = models.CharField(verbose_name=_('title'), max_length=300)
+    originator_email = models.CharField(verbose_name=_('email'), max_length=300)
+    originator_phone = models.CharField(verbose_name=_('phone'), max_length=50, blank=True)
+
+    ifrc_name = models.CharField(verbose_name=_('name'), max_length=100)
+    ifrc_title = models.CharField(verbose_name=_('title'), max_length=300)
+    ifrc_email = models.CharField(verbose_name=_('email'), max_length=300)
+    ifrc_phone = models.CharField(verbose_name=_('phone'), max_length=50, blank=True)
```
Comment (k9845) [inline informal_update/models.py:80]: lets make this null=True, blank=True

File: CHANGELOG.md [hunk 11]
```diff
@@ -6,6 +6,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## Unreleased
 
+## 1.1.413
+
+### Added
+ - Improved email layout for new operations (=appeals)
+ - Ingest appeals - process all, regardless the modify time
+
```
Comment (k9845) [inline CHANGELOG.md:14]: why are these generated?

File: CHANGELOG.md [hunk 12]
```diff
@@ -1856,7 +1862,8 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 ## 0.1.20
 
-[Unreleased]: https://github.com/IFRCGo/go-api/compare/1.1.412...HEAD
+[Unreleased]: https://github.com/IFRCGo/go-api/compare/1.1.413...HEAD
+[1.1.413]: https://github.com/IFRCGo/go-api/compare/1.1.412...1.1.413
```
Comment (k9845) [inline CHANGELOG.md:1866]: Same here as well

File: main/__init__.py [hunk 13]
```diff
@@ -3,4 +3,4 @@
 from .celery import app as celery_app
 
 __all__ = ['celery_app']
-__version__ = '1.1.412'
+__version__ = '1.1.413'
```
Comment (k9845) [inline main/__init__.py:6]: Is this updated?

**Review by k9845 — CHANGES_REQUESTED** [review]

File: api/models.py (hunk 2 above)
Comment (Rup-Narayan-Rajbanshi) [reply api/models.py:1385]: We have one more option Government in the mockup, so it is added.

File: informal_update/models.py (hunk 9 above)
Comment (Rup-Narayan-Rajbanshi) [reply informal_update/models.py:83]: For now let us keep default( recommended )

**Pull request closed without merging**